libs/platform/experience/carousel-navigation/carousel-navigation.component.ts

```diff
@@ -0,0 +1,410 @@
+{
+  static styles = carouselNavigationStyles;
+
+  @property({ type: Boolean }) showArrows?: boolean;
+  @property({ type: Boolean }) showIndicators?: boolean;
+  @property() arrowNavigationBehavior?: ArrowNavigationBehavior;
+  @property({ reflect: true }) indicatorsPosition?: CarouselIndicatorPosition;
+  @property({ reflect: true }) indicatorsAlignment?: CarouselIndicatorAlignment;
+  @property() scrollBehavior?: CarouselScrollBehavior;
+
+  protected resizeObserver?: ResizeObserver;
+  protected intersectionObserver?: IntersectionObserver;
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  protected throttledScrollListener: () => void = () => {};
+
+  @state() protected items: HTMLElement[] = [];
+  @state() protected slides: { index: number }[] = [];
+
+  @queryAll('input') protected indicatorElements?: NodeListOf<HTMLInputElement>;
+
+  /**
+   * @override Initialize the intersection observer. The intersection observer is
+   * used to detect when the carousel becomes visible and sets the resize observer
+   * and listens to scroll events to always keep the state up to date.
+   */
+  connectedCallback(): void {
+    this.initializeIntersectionObserver();
+    super.connectedCallback();
+  }
+
+  disconnectedCallback(): void {
+    this.resizeObserver?.disconnect();
+    this.intersectionObserver?.disconnect();
+    if (this.hostElement) {
+      this.hostElement.removeEventListener(
+        'scroll',
+        this.throttledScrollListener
+      );
+      this.hostElement.removeEventListener('scrollend', this.updateState);
+    }
+    super.disconnectedCallback();
+  }
+
+  /**
+   * Initializes the IntersectionObserver to detect when the carousel becomes visible.
+   * This is necessary to calculate the carousel's width and to build the navigation.
+   */
+  protected initializeIntersectionObserver(): void {
+    if (this.intersectionObserver) return;
+
+    this.intersectionObserver = new IntersectionObserver(
+      throttle((entries: IntersectionObserverEntry[]) => {
+        return entries.forEach((entry) => {
+          if (entry.isIntersecting) {
+            if (this.resolveItems().length) {
+              this.buildNavigation();
+              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+              this.intersectionObserver!.disconnect();
+              this.initializeScrollListener();
+              this.initializeResizeObserver();
+            }
+          }
+        });
+      }, 150),
+      { root: null, rootMargin: '0px', threshold: 1.0 }
+    );
+    this.intersectionObserver.observe(this.hostElement);
+  }
+
+  /**
+   * Initializes the ResizeObserver to detect when the carousel's width changes.
+   * This is necessary to recalculate the carousel's width and build the navigation.
+   */
+  protected initializeResizeObserver(): void {
+    if (this.resizeObserver) return;
+    const throttleTime = this.hostElement.clientWidth === 0 ? 0 : 150;
```

Comment:
I would store the default throttle in prop or const, ie:
`const throttleTime = this.hostElement.clientWidth ? defaultThrottleTime: 0;`